source/speech/speech.py

```diff
@@ -1169,9 +1171,34 @@ def speak(  # noqa: C901
 			)
 			if not inCharacterMode:
 				speechSequence[index] += CHUNK_SEPARATOR
+				if (
+					config.conf["speech"]["reportLanguage"]
+					and curLanguage not in (defaultLanguage, _speechState.lastReportedLanguage)
+					and languageHandler.getLanguageDescription(curLanguage) not in speechSequence
```

Comment:
What happens if you read an English sentence containing two French words?
For example in: <span lang="fr">bonjour</div> and <span lang="fr">bonsoir</div> in a row? Are the two language changes reported?

Reply:
> @nvdaes - can you please answer this question from @CyrilleB79 [#17685 (comment)](https://github.com/nvaccess/nvda/pull/17685#discussion_r1989578239)


@CyrilleB79 , yes, you can test your own example please.